# 0. import 라이브러리 


In [1]:
import numpy as np
import pandas as pd

# 1. 데이터 준비

## 1.1 데이터 불러오기

In [2]:
# 데이터 불러오기
recruitment_data = pd.read_csv("./data/raw_data/recruitment.csv")
company_data = pd.read_csv("./data/raw_data/company.csv")
resume_data = pd.read_csv("./data/raw_data/resume.csv")
resume_certificate_data = pd.read_csv("./data/raw_data/resume_certificate.csv")
resume_education_data = pd.read_csv("./data/raw_data/resume_education.csv")
resume_language_data = pd.read_csv("./data/raw_data/resume_language.csv")

apply_train_data = pd.read_csv("./data/raw_data/apply_train.csv")
sample_submission_data = pd.read_csv("./data/raw_data/sample_submission.csv")

## 1.2 데이터 확인 및 정제

### 1.2.1 recruitment (6,695)

<details>
<summary>  <span style="color:gold"> 상세정보 펼쳐보기 </span>  </summary>

</br>

|seq|columns|description|note|
|--|--|--|--|
|0|recruitment_seq|공고번호|PK|
|1|address_seq1|근무지주소코드1||
|2|address_seq2|근무지주소코드2||
|3|address_seq3|근무지주소코드3||
|4|career_end|요구 최대경력||
|5|career_start|요구 최소경력||
|6|check_box_keyword|모집직무코드|; 으로 구분|
|7|education|요구학위||
|8|major_task|주업무코드||
|9|qualifications|자격요건난이도||
|10|text_keyword|모집직무키워드|; 으로 구분|

</details>

In [3]:
# 데이터 확인
display(recruitment_data.tail(3))

# 결측치 확인
print(recruitment_data.isna().sum())

recruitment_seq  address_seq1  address_seq2  address_seq3  career_end  \
6692          R03252           3.0           NaN           NaN           0   
6693          R05130           3.0           NaN           NaN           0   
6694          R04779           3.0           NaN           NaN           0   

      career_start    check_box_keyword  education  major_task  \
6692             0                 2109          3           2   
6693             0  2201;2204;2205;2707          2           2   
6694             0  2101;2108;2201;2707          3           8   

      qualifications text_keyword  
6692               1          NaN  
6693               2          NaN  
6694               1          NaN

recruitment_seq         0
address_seq1            1
address_seq2         6595
address_seq3         6686
career_end              0
career_start            0
check_box_keyword       0
education               0
major_task              0
qualifications          0
text_keyword         5988
dtype: int64


In [4]:
# address_seq 컬럼 중 NaN를 0.0로 변경
recruitment_data.address_seq1 = recruitment_data.address_seq1.replace(np.NAN, 0.0)
recruitment_data.address_seq2 = recruitment_data.address_seq2.replace(np.NAN, 0.0)
recruitment_data.address_seq3 = recruitment_data.address_seq3.replace(np.NAN, 0.0)

# text_keyword 컬럼 중 NaN를 'NaN'로 변경
recruitment_data.text_keyword = recruitment_data.text_keyword.replace(np.NAN, 'NaN')

# 결측치 재확인
print(recruitment_data.isna().sum())

recruitment_seq      0
address_seq1         0
address_seq2         0
address_seq3         0
career_end           0
career_start         0
check_box_keyword    0
education            0
major_task           0
qualifications       0
text_keyword         0
dtype: int64


### 1.2.2 company (2,377)

<details>
<summary>  <span style="color:gold"> 상세정보 펼쳐보기 </span>  </summary>

</br>

|seq|columns|description|note|
|--|--|--|--|
|0|recruitment_seq|공고번호|FK|
|1|company_type_seq|회사유형코드||
|2|supply_kind|주업종코드||
|3|employee|종업원수||
<details>

In [5]:
# 데이터 확인
display(company_data.tail(3))

# 결측치 확인
print(company_data.isna().sum())

recruitment_seq  company_type_seq  supply_kind  employee
2374          R04028                 4          402       525
2375          R06508                 2          402        70
2376          R00604                 5          402       300

recruitment_seq     0
company_type_seq    0
supply_kind         0
employee            0
dtype: int64


### 1.2.3 resume (8,482)

<details>
<summary>  <span style="color:gold"> 상세정보 펼쳐보기 </span>  </summary>

</br>

|seq|columns|description|note|
|--|--|--|--|
|0|resume_seq|이력서번호|PK|
|1|reg_data|이력서등록일||
|2|updated_data|이력서수정일||
|3|degree|최종학력||
|4|graduate_data|졸업년도||
|5|hope_salary|희망연봉||
|6|last_salary|최종연봉||
|7|text_keyword|직무키워드|; 으로 구분|
|8|job_code_seq1|희망직무1||
|9|job_code_seq2|희망직무2||
|10|job_code_seq3|희망직무3||
|11|career_month|경력월||
|12|career_job_code|경력직무|; 으로 구분|
</details>

In [6]:
# 데이터 확인
display(resume_data.tail(3))

# 결측치 확인
print(resume_data.isna().sum())

resume_seq    reg_date updated_date  degree  graduate_date  hope_salary  \
8479     U01468  2019-02-26   2020-06-23       4           2008          0.0   
8480     U05315  2020-03-27   2020-12-21       4           2016          0.0   
8481     U01277  2020-09-22   2020-09-22       4           2009       2100.0   

      last_salary          text_keyword job_code_seq1 job_code_seq2  \
8479          0.0          상품기획MD;우븐디자인           디자인           NaN   
8480          0.0  여성복디자인;우븐디자인;다이마루디자인   재료·화학·섬유·의복           NaN   
8481       1000.0            상품기획;머천다이저   재료·화학·섬유·의복           NaN   

     job_code_seq3  career_month career_job_code  
8479           NaN             0     섬유;봉제;가방;의류  
8480           NaN            18      디자인;CAD;설계  
8481           NaN             3     섬유;봉제;가방;의류

resume_seq            0
reg_date              0
updated_date          0
degree                0
graduate_date         0
hope_salary           0
last_salary           0
text_keyword        195
job_code_seq1         0
job_code_seq2      7982
job_code_seq3      8295
career_month          0
career_job_code     758
dtype: int64


In [7]:
# text_keyword 컬럼 중 NaN를 'NaN'로 변경
resume_data.text_keyword = resume_data.text_keyword.replace(np.NAN, 'NaN')

# job_code_seq 컬럼 중 NaN를 'NaN'로 변경
resume_data.job_code_seq2 = resume_data.job_code_seq2.replace(np.NAN, 'NaN')
resume_data.job_code_seq3 = resume_data.job_code_seq3.replace(np.NAN, 'NaN')

# career_job_code 컬럼 중 NaN를 'NaN'로 변경
resume_data.career_job_code = resume_data.career_job_code.replace(np.NAN, 'NaN')


# 결측치 재확인
print(recruitment_data.isna().sum())

recruitment_seq      0
address_seq1         0
address_seq2         0
address_seq3         0
career_end           0
career_start         0
check_box_keyword    0
education            0
major_task           0
qualifications       0
text_keyword         0
dtype: int64


### 1.2.4 resume_certificate (12,975)

<details>
<summary>  <span style="color:gold"> 상세정보 펼쳐보기 </span>  </summary>

</br>

|seq|columns|description|note|
|--|--|--|--|
|0|resume_seq|이력서번호|FK|
|1|certificate_contents|자격증명||
</details>

In [8]:
# 데이터 확인
display(resume_certificate_data.tail(3))

# 결측치 확인
print(resume_certificate_data.isna().sum())

resume_seq certificate_contents
12972     U03248            수치제어밀링기능사
12973     U08157                 주산2급
12974     U02889             유통관리사 2급

resume_seq              0
certificate_contents    4
dtype: int64


In [9]:
# certificate_contents 컬럼 중 NaN를 'NaN'로 변경
resume_certificate_data.certificate_contents = resume_certificate_data.certificate_contents.replace(np.NAN, 'NaN')


# 결측치 재확인
print(resume_certificate_data.isna().sum())

resume_seq              0
certificate_contents    0
dtype: int64


### 1.2.5 resume_education (8,482)

<details>
<summary>  <span style="color:gold"> 상세정보 펼쳐보기 </span>  </summary>

</br>

|seq|columns|description|note|
|--|--|--|--|
|0|resume_seq|이력서번호|FK|
|1|hischool_type_seq|고등학교유형코드||
|2|hischool_special_type|특수고등학교분류||
|3|hischool_nation|고등학교공립/사립구분||
|4|hischool_gender|고등학교남녀/공학구분||
|5|hischool_location_seq|고등학교지역코드||
|6|univ_type_seq1|최종대학(원)유형코드1||
|7|univ_type_seq2|최종대학(원)유형코드2||
|8|univ_tranfer|최종대학(원)편입여부|편입: 1|
|9|univ_location|최종대학(원)지역코드|
|10|univ_major|최종대학(원)전공||
|11|univ_sub_major|최종대학(원)부전공||
|12|univ_major_type|최종대학(원)주전공계열코드||
|13|univ_score|최종대학(원)평균평점||
</details>

In [10]:
# 데이터 확인
display(resume_education_data.tail(3))

# 결측치 확인
print(resume_education_data.isna().sum())

resume_seq  hischool_type_seq hischool_special_type hischool_nation  \
8479     U04109                 21                   일반고              공립   
8480     U01863                 21                   일반고              사립   
8481     U05884                  0                    기타              기타   

     hischool_gender  hischool_location_seq  univ_type_seq1  univ_type_seq2  \
8479            남여공학                      5               0               0   
8480            여자학교                     15               3               5   
8481              기타                      0               5               5   

      univ_transfer  univ_location  univ_major univ_sub_major  \
8479              0              0         NaN            NaN   
8480              0             15  의류학과생활과학계열            NaN   
8481              0             11         NaN            NaN   

      univ_major_type  univ_score  
8479                9        80.0  
8480                9        80.0  
8481                9        90.0

resume_seq                  0
hischool_type_seq           0
hischool_special_type       0
hischool_nation             0
hischool_gender             0
hischool_location_seq       0
univ_type_seq1              0
univ_type_seq2              0
univ_transfer               0
univ_location               0
univ_major               6687
univ_sub_major           8066
univ_major_type             0
univ_score                  0
dtype: int64


In [11]:
# certificate_contents 컬럼 중 NaN를 'NaN'로 변경
resume_education_data.univ_major  = resume_education_data.univ_major .replace(np.NAN, 'NaN')

# certificate_contents 컬럼 중 NaN를 'NaN'로 변경
resume_education_data.univ_sub_major  = resume_education_data.univ_sub_major.replace(np.NAN, 'NaN')


# 결측치 재확인
print(resume_education_data.isna().sum())

resume_seq               0
hischool_type_seq        0
hischool_special_type    0
hischool_nation          0
hischool_gender          0
hischool_location_seq    0
univ_type_seq1           0
univ_type_seq2           0
univ_transfer            0
univ_location            0
univ_major               0
univ_sub_major           0
univ_major_type          0
univ_score               0
dtype: int64


### 1.2.6 resume_language (869)

<details>
<summary>  <span style="color:gold"> 상세정보 펼쳐보기 </span>  </summary>

</br>

|seq|columns|description|note|
|--|--|--|--|
|0|resume_seq|이력서번호|FK|
|1|language|외국어코드||
|2|exam_name|공인시험코드||
|3|score|점수||
</details>

In [12]:
# 데이터 확인
display(resume_language_data.tail(3))

# 결측치 확인
print(resume_language_data.isna().sum())

resume_seq  language  exam_name   score
866     U05922         2          2  500.00
867     U03827         2          4  883.84
868     U04680         2          4  666.67

resume_seq    0
language      0
exam_name     0
score         0
dtype: int64


### 1.2.7 apply_train (57,946)

<details>
<summary>  <span style="color:gold"> 상세정보 펼쳐보기 </span>  </summary>

</br>

|seq|columns|description|note|
|--|--|--|--|
|0|resume_seq|이력서번호|FK|
|<span style="color:yellow">1|<span style="color:yellow">recruitment_seq|<span style="color:yellow">공고번호|<span style="color:yellow">TARGET(FK)</span>|
</deatils>

In [13]:
# 데이터 확인
display(apply_train_data.tail(3))

# 결측치 확인
print(apply_train_data.isna().sum())

resume_seq recruitment_seq
57943     U08238          R01342
57944     U01296          R06363
57945     U05748          R03090

resume_seq         0
recruitment_seq    0
dtype: int64


### 1.2.8 sample_submission (42,410)

> 사용자별로 5개씩 공고ID추천리스트 생성

<details>
<summary>  <span style="color:gold"> 상세정보 펼쳐보기 </span>  </summary>

</br>

|seq|columns|description|note|
|--|--|--|--|
|0|resume_seq|이력서번호|FK|
|<span style="color:yellow">1|<span style="color:yellow">recruitment_seq|<span style="color:yellow">공고번호|<span style="color:yellow">TARGET(FK)</span>|
</details>

In [14]:
# 데이터 확인
display(sample_submission_data.tail(3))

# 결측치 확인
print(sample_submission_data.isna().sum())

resume_seq recruitment_seq
42407     U08482          R01931
42408     U08482          R01607
42409     U08482          R06059

resume_seq         0
recruitment_seq    0
dtype: int64


## 1.3 데이터 전처리

### 1.3.1 채용 공고 관련 데이터 중 사용할 데이터 추출

<details>
<summary>  <span style="color:gold"> 상세정보 펼쳐보기 </span>  </summary>

</br>

|seq|columns|description|example|note|
|--|--|--|--|--|
|0|<span style="color:orange">recruitment_seq|<span style="color:orange">공고번호|R02444|PK|
|1|<span style="color:orange">address_seq1|<span style="color:orange">근무지주소코드1|3.0||
|2|<span style="color:orange">address_seq2|<span style="color:orange">근무지주소코드2|20.0||
|3|address_seq3|근무지주소코드3|20.0|기입된 데이터는 9개 뿐으로 사용 X|
|4|career_end|요구 최대경력|0|모든 값이 0|
|5|career_start|요구 최소경력|0|모든 값이 0|
|6|<span style="color:orange">check_box_keyword|<span style="color:orange">모집직무코드|2507;2707;2810|; 으로 구분|
|7|<span style="color:orange">education|<span style="color:orange">요구학위|4||
|8|<span style="color:orange">major_task|<span style="color:orange">주업무코드|8||
|9|<span style="color:orange">qualifications|<span style="color:orange">자격요건난이도|1||
|10|text_keyword|모집직무키워드|MD;기획MD;여성복;캐주얼|; 으로 구분 / 707개의 데이터가 있지만 확인 결과 'check_box_keyword'컬럼이 더 객관적 분류 가능|
|11|<span style="color:orange">company_type_seq|<span style="color:orange">회사유형코드|2||
|12|<span style="color:orange">supply_kind|<span style="color:orange">주업종코드|514||
|13|<span style="color:orange">employee|<span style="color:orange">종업원수|20||
</details>

In [15]:
# 필요 데이터 추출
rec_data = pd.merge(recruitment_data.drop(['address_seq3','career_end','career_start','text_keyword'], axis='columns'), company_data, how='outer', on='recruitment_seq')
display(rec_data.tail(3))

recruitment_seq  address_seq1  address_seq2    check_box_keyword  \
6692          R03252           3.0           0.0                 2109   
6693          R05130           3.0           0.0  2201;2204;2205;2707   
6694          R04779           3.0           0.0  2101;2108;2201;2707   

      education  major_task  qualifications  company_type_seq  supply_kind  \
6692          3           2               1               4.0        402.0   
6693          2           2               2               2.0        402.0   
6694          3           8               1               2.0        402.0   

      employee  
6692     525.0  
6693      40.0  
6694      41.0

In [16]:
# 결측치가 있는 컬럼명 출력
print(', '.join([col for col in rec_data.columns if rec_data[col].isna().sum() != 0]))

company_type_seq, supply_kind, employee


In [17]:
# 데이터 정제 및 수정
rec_data.company_type_seq = rec_data.company_type_seq.replace(np.NAN, 0)
rec_data.supply_kind = rec_data.supply_kind.replace(np.NAN, 0)
rec_data.employee = rec_data.employee.replace(np.NAN, 0)

# 결측치 재확인
print(rec_data.isna().sum())

recruitment_seq      0
address_seq1         0
address_seq2         0
check_box_keyword    0
education            0
major_task           0
qualifications       0
company_type_seq     0
supply_kind          0
employee             0
dtype: int64


In [18]:
# 완성 데이터 저장
# rec_data.to_csv('./data/pre-processed_data/rec_data.csv', encoding='utf-8-sig')

### 1.3.2 이력서 관련 데이터 중 사용할 데이터 추출

<details>
<summary> <span style="color:gold"> 상세정보 펼쳐보기 </span> </summary>

</br>

|seq|columns|description|example|note|
|--|--|--|--|--|
|0|<span style="color:orange">resume_seq|<span style="color:orange">이력서번호|U01468|PK|
|1|reg_data|이력서등록일|2019-02-26||
|2|updated_data|이력서수정일|2020-06-23||
|3|<span style="color:orange">degree|<span style="color:orange">최종학력|4||
|4|graduate_data|졸업년도|2008||
|5|<span style="color:orange">hope_salary|<span style="color:orange">희망연봉|2100.0||
|6|last_salary|최종연봉|1000.0||
|7|<span style="color:orange">text_keyword|<span style="color:orange">직무키워드|상품기획MD;우븐디자인|; 으로 구분|
|8|<span style="color:green">job_code_seq1|<span style="color:green">희망직무1|경영·기획·회계·사무|희망직무 1~3은 순서에 상관없음으로 확인|
|9|<span style="color:green">job_code_seq2|<span style="color:green">희망직무2|재료·화학·섬유·의복|>> 희망직무1~3 를 기존의 10개의 직무를 바탕으로 새로운 분류 코드 생성|
|10|<span style="color:green">job_code_seq3|<span style="color:green">희망직무3|영업·판매·TM|>> '기타 직종'과 'NaN'은 '기타 직종' / '여행사'는 '관광'으로 통합|
|11|<span style="color:orange">career_month|<span style="color:orange">경력월|120|1년 단위로 그룹화|
|12|<span style="color:orange">career_job_code|<span style="color:orange">경력직무|섬유;봉제;가방;의류|; 으로 구분|
|13|certificate_contents|자격증명|주산2급||
|14|hischool_type_seq|고등학교유형코드|21|사용 X (최종학력으로만 구분 예정)|
|15|hischool_special_type|특수고등학교분류|일반고|사용 X (최종학력으로만 구분 예정)|
|16|hischool_nation|고등학교공립/사립구분|사립|사용 X (최종학력으로만 구분 예정)|
|17|hischool_gender|고등학교남녀/공학구분|여자학교|사용 X (최종학력으로만 구분 예정)|
|18|hischool_location_seq|고등학교지역코드|15|사용 X (최종학력으로만 구분 예정)|
|19|<span style="color:orange">univ_type_seq1|<span style="color:orange">최종대학(원)유형코드1|3|6종류|
|20|<span style="color:orange">univ_type_seq2|<span style="color:orange">최종대학(원)유형코드2|5|10종류|
|21|univ_tranfer|최종대학(원)편입여부|0|편입: 1|
|22|univ_location|최종대학(원)지역코드|15||
|23|univ_major|최종대학(원)전공|의류학과생활과학계열|주전공계열코드 사용|
|24|<span style="color:orange">univ_sub_major|<span style="color:orange">최종대학(원)부전공|의상디자인교육학||
|25|<span style="color:orange">univ_major_type|<span style="color:orange">최종대학(원)주전공계열코드|9||
|26|<span style="color:orange">univ_score|<span style="color:orange">최종대학(원)평균평점|80.0||
|27|<span style="color:green">language|<span style="color:green">외국어코드|2|총 5종류|
|28|<span style="color:green">exam_name|<span style="color:green">공인시험코드|4|총 15종류|
|29|<span style="color:green">score|<span style="color:green">점수|883.84|가산점 형식으로 외국어 점수 계산; 점수에 따라 초급,중급,고급으로 분류(상대평가) 후 보유외국어 실력만큼 점수 추가|

</details>

In [132]:
# 희망직무 통합 코드 만들기
## resume_data 중 사용할 컬럼 가져오기
resume_temp = resume_data[['resume_seq','degree','hope_salary','text_keyword','job_code_seq1','job_code_seq2','job_code_seq3','career_month','career_job_code']]

## 통합 12개의 희망직무 코드 중 선택한 직무 코드를 하나로 묶어주기
resume_temp = resume_temp.replace({'job_code_seq2':'여행사'},'관광').replace({'job_code_seq2':'NaN'},'기타 직종').replace({'job_code_seq3':'NaN'},'기타 직종')
new_job_code_seq = {'IT·게임':'a', '건설·기계·전기·전자':'b','경영·기획·회계·사무':'c', '관광':'d', '기술·과학·산업':'e', '디자인':'f', '문화·예술·신문·방송':'g', '영업·판매·TM':'h', '재료·화학·섬유·의복':'i', '전문·교육·자격':'j', '통신·모바일':'k', '기타 직종':''}

resume_temp['job_code_seq'] = ''
for i in range(len(resume_temp)):
    resume_temp.loc[i,'job_code_seq'] = ''.join(sorted([new_job_code_seq[resume_temp.loc[i]['job_code_seq1']], new_job_code_seq[resume_temp.loc[i]['job_code_seq2']],new_job_code_seq[resume_temp.loc[i]['job_code_seq3']]]))

## 기존의 희망직무1~3 제거
resume_temp = resume_temp.drop(['job_code_seq1','job_code_seq2','job_code_seq3'], axis='columns')

resume_seq  degree  hope_salary          text_keyword  career_month  \
8479     U01468       4          0.0          상품기획MD;우븐디자인             0   
8480     U05315       4          0.0  여성복디자인;우븐디자인;다이마루디자인            18   
8481     U01277       4       2100.0            상품기획;머천다이저             3   

     career_job_code job_code_seq  
8479     섬유;봉제;가방;의류            f  
8480      디자인;CAD;설계            i  
8481     섬유;봉제;가방;의류            i

In [ ]:
# 1000(만원)단위로 희망 연봉분류 

In [ ]:
# 이력서 통합 데이터 (res_data)로 병합
res_data = resume_temp
display(res_data.tail(3))

In [136]:
max(res_data.hope_salary)
# len(resume_temp.job_code_seq.value_counts())

10000.0

In [121]:
resume_temp[resume_temp.job_code_seq == '']

resume_seq  degree  hope_salary text_keyword job_code_seq1 job_code_seq2  \
2856     U07242       4          0.0          기타직         기타 직종         기타 직종   

     job_code_seq3  career_month career_job_code job_code_seq  
2856         기타 직종           108          기타 서비스

'IT·게임', '건설·기계·전기·전자','경영·기획·회계·사무', '관광', '기술·과학·산업', '디자인', '문화·예술·신문·방송', '영업·판매·TM', '재료·화학·섬유·의복', '전문·교육·자격', '통신·모바일', '기타 직종'

<br>
['IT·게임', '건설·기계·전기·전자', '경영·기획·회계·사무', '기타 직종', '디자인', '문화·예술·신문·방송', '영업·판매·TM', '재료·화학·섬유·의복', '전문·교육·자격']<br>
['IT·게임', 'NaN', '건설·기계·전기·전자', '경영·기획·회계·사무', '기술·과학·산업', '기타 직종', '디자인', '문화·예술·신문·방송', '여행사', '영업·판매·TM', '재료·화학·섬유·의복', '전문·교육·자격']<br>
['IT·게임', 'NaN', '건설·기계·전기·전자', '관광', '기술·과학·산업', '기타 직종', '디자인', '문화·예술·신문·방송', '영업·판매·TM', '재료·화학·섬유·의복', '전문·교육·자격', '통신·모바일']

In [118]:
display(resume_temp.tail(3))

resume_seq  degree  hope_salary          text_keyword job_code_seq1  \
8479     U01468       4          0.0          상품기획MD;우븐디자인           디자인   
8480     U05315       4          0.0  여성복디자인;우븐디자인;다이마루디자인   재료·화학·섬유·의복   
8481     U01277       4       2100.0            상품기획;머천다이저   재료·화학·섬유·의복   

     job_code_seq2 job_code_seq3  career_month career_job_code job_code_seq  
8479         기타 직종         기타 직종             0     섬유;봉제;가방;의류            f  
8480         기타 직종         기타 직종            18      디자인;CAD;설계            i  
8481         기타 직종         기타 직종             3     섬유;봉제;가방;의류            i

In [66]:
res_data = pd.merge()

4    7453
3     463
5     349
2     211
6       6
Name: degree, dtype: int64

### momo

In [67]:
# display(resume_data[resume_data.degree == 2].tail(5))
# display(resume_data[resume_data.degree == 3].tail(5))
# display(resume_data[resume_data.degree == 4].tail(5))
# display(resume_data[resume_data.degree == 5].tail(5))
# display(resume_data[resume_data.degree == 6].tail(5))

In [68]:
#2
resume_education_data[(resume_education_data.resume_seq =='U02566') | (resume_education_data.resume_seq =='U03951') | (resume_education_data.resume_seq =='U03746') | (resume_education_data.resume_seq =='U07065') | (resume_education_data.resume_seq =='U00979')]

resume_seq  hischool_type_seq hischool_special_type hischool_nation  \
271      U03951                 21                   일반고              공립   
436      U03746                 21                   일반고              공립   
3009     U00979                 16                  특성화고              사립   
4335     U07065                 21                   일반고              사립   
6800     U02566                 21                   일반고              사립   

     hischool_gender  hischool_location_seq  univ_type_seq1  univ_type_seq2  \
271             남여공학                      5               5               5   
436             남자학교                      3               5               5   
3009            남여공학                      3               5               5   
4335            남여공학                     17               3               6   
6800            남자학교                      3               5              10   

      univ_transfer  univ_location univ_major univ_sub_major  univ_major_type  \
271               0              3      미디어정보            NaN                4   
436               0              5     신학연구학과            NaN               21   
3009              0              3        NaN            NaN                3   
4335              0              3        NaN            NaN               18   
6800              0              5        NaN            NaN                4   

      univ_score  
271         80.0  
436          0.0  
3009         0.0  
4335         0.0  
6800         0.0

In [69]:
#3
resume_education_data[(resume_education_data.resume_seq =='U07442') | (resume_education_data.resume_seq =='U00425') | (resume_education_data.resume_seq =='U04458') | (resume_education_data.resume_seq =='U01990') | (resume_education_data.resume_seq =='U04770')]

resume_seq  hischool_type_seq hischool_special_type hischool_nation  \
1307     U00425                 21                   일반고              사립   
2041     U01990                  0                    기타              기타   
5685     U04458                 21                   일반고              공립   
6188     U07442                 22                   일반고              사립   
7801     U04770                 21                   일반고              공립   

     hischool_gender  hischool_location_seq  univ_type_seq1  univ_type_seq2  \
1307            남자학교                      3               5              10   
2041              기타                      5               5              10   
5685            남여공학                     17               5              10   
6188            여자학교                      5               0               0   
7801            여자학교                     15               5               7   

      univ_transfer  univ_location univ_major univ_sub_major  univ_major_type  \
1307              0              5   컴퓨터산업디자인            NaN                9   
2041              0              5        NaN            NaN                5   
5685              0             17        NaN            NaN                7   
6188              0              0        NaN            NaN                9   
7801              0              3        NaN            NaN                5   

      univ_score  
1307        80.0  
2041        80.0  
5685        60.0  
6188        20.0  
7801        20.0

In [70]:
#4
resume_education_data[(resume_education_data.resume_seq =='U08092') | (resume_education_data.resume_seq =='U05910') | (resume_education_data.resume_seq =='U01468') | (resume_education_data.resume_seq =='U05315') | (resume_education_data.resume_seq =='U01277')]

resume_seq  hischool_type_seq hischool_special_type hischool_nation  \
533      U08092                 21                   일반고              공립   
833      U05910                 21                   일반고              공립   
4728     U05315                 21                   일반고              공립   
4924     U01277                 21                   일반고              공립   
8186     U01468                  0                    기타              기타   

     hischool_gender  hischool_location_seq  univ_type_seq1  univ_type_seq2  \
533             남여공학                      3               5               5   
833             여자학교                     15               5               5   
4728            남여공학                      5               5               5   
4924            여자학교                      5               5               5   
8186              기타                      0               5               5   

      univ_transfer  univ_location univ_major univ_sub_major  univ_major_type  \
533               0              5        NaN            NaN                0   
833               0             15        NaN            NaN               19   
4728              0              3        NaN            NaN                9   
4924              0              3        NaN            NaN                5   
8186              0             18        NaN            NaN                9   

      univ_score  
533         80.0  
833         80.0  
4728        80.0  
4924        80.0  
8186        80.0

In [71]:
#5
resume_education_data[(resume_education_data.resume_seq =='U08220') | (resume_education_data.resume_seq =='U03215') | (resume_education_data.resume_seq =='U03836') | (resume_education_data.resume_seq =='U02341') | (resume_education_data.resume_seq =='U04890')]

resume_seq  hischool_type_seq hischool_special_type hischool_nation  \
215      U08220                 21                   일반고              사립   
252      U03215                 21                   일반고              사립   
953      U02341                 21                   일반고              사립   
2457     U03836                 16                  특성화고              사립   
7663     U04890                 21                   일반고              사립   

     hischool_gender  hischool_location_seq  univ_type_seq1  univ_type_seq2  \
215             남자학교                      3               5               5   
252             여자학교                      3               5               5   
953             여자학교                     11               3               5   
2457            여자학교                      8               5               5   
7663            남자학교                      3               5               5   

      univ_transfer  univ_location univ_major univ_sub_major  univ_major_type  \
215               0              4        NaN            NaN                9   
252               0              3        NaN            NaN                9   
953               0             15        NaN            NaN               19   
2457              0              8        NaN            NaN                9   
7663              0              3        NaN            NaN                3   

      univ_score  
215         80.0  
252         70.0  
953         90.0  
2457        90.0  
7663        70.0

In [72]:
#6
resume_education_data[(resume_education_data.resume_seq =='U03051') | (resume_education_data.resume_seq =='U07334') | (resume_education_data.resume_seq =='U05886') | (resume_education_data.resume_seq =='U03595') | (resume_education_data.resume_seq =='U08309')]

resume_seq  hischool_type_seq hischool_special_type hischool_nation  \
1192     U03595                 21                   일반고              사립   
2371     U03051                 21                   일반고              사립   
7968     U07334                 21                   자율고              사립   
8282     U08309                 21                   일반고              사립   
8413     U05886                  0                    기타              기타   

     hischool_gender  hischool_location_seq  univ_type_seq1  univ_type_seq2  \
1192            여자학교                      3               5               5   
2371            여자학교                     13               5               5   
7968            남자학교                      3               0               0   
8282            여자학교                      8               0               0   
8413              기타                      0               5               5   

      univ_transfer  univ_location                 univ_major univ_sub_major  \
1192              0             10                       영어영문            NaN   
2371              0             18                        NaN            NaN   
7968              0              0                      패션경영학            NaN   
8282              0              0  PatternmakingofTechnology            NaN   
8413              0              3                        NaN            NaN   

      univ_major_type  univ_score  
1192                0        80.0  
2371                9        90.0  
7968                3        80.0  
8282                4        80.0  
8413                0        70.0

In [73]:
display(resume_education_data[(resume_education_data.hischool_type_seq == 0) & (resume_education_data.univ_type_seq1 == 0) & (resume_education_data.univ_score == 0)])
# resume_education_data[resume_education_data.univ_sub_major == 'NaN']

resume_seq  hischool_type_seq hischool_special_type hischool_nation  \
2096     U06891                  0                    기타              기타   
2107     U06287                  0                    기타              기타   
2352     U05363                  0                    기타              기타   
2855     U02426                  0                    기타              기타   
2981     U05781                  0                    기타              기타   
3846     U03451                  0                    기타              기타   
4606     U06734                  0                    기타              기타   
4950     U08267                  0                    기타              기타   
5667     U04907                  0                    기타              기타   
5738     U03170                  0                    기타              기타   
5857     U07989                  0                    기타              기타   
6471     U06675                  0                    기타              기타   
6706     U03662                  0                    기타              기타   
6908     U01951                  0                    기타              기타   
6977     U02145                  0                    기타              기타   
7351     U02068                  0                    기타              기타   
7419     U04640                  0                    기타              기타   
7629     U05940                  0                    기타              기타   
7799     U00538                  0                    기타              기타   
7875     U04417                  0                    기타              기타   
7916     U05522                  0                    기타              기타   
8037     U04830                  0                    기타              기타   
8217     U03561                  0                    기타              기타   

     hischool_gender  hischool_location_seq  univ_type_seq1  univ_type_seq2  \
2096              기타                      0               0               0   
2107              기타                      0               0               0   
2352              기타                      0               0               0   
2855              기타                      3               0               0   
2981              기타                      0               0               0   
3846              기타                      0               0               0   
4606              기타                      0               0               0   
4950              기타                      0               0               0   
5667              기타                      0               0               0   
5738              기타                      0               0               0   
5857              기타                      0               0               0   
6471              기타                      0               0               0   
6706              기타                      0               0               0   
6908              기타                      3               0               0   
6977              기타                      0               0               0   
7351              기타                      0               0               0   
7419              기타                      0               0               0   
7629              기타                      5               0               0   
7799              기타                      0               0               0   
7875              기타                      0               0               0   
7916              기타                      0               0               0   
8037              기타                      0               0               0   
8217              기타                      0               0               0   

      univ_transfer  univ_location                univ_major  \
2096              0              0                       NaN   
2107              0              0                       NaN   
2352              0              0                       NaN   
2855              0        

In [74]:
print(resume_education_data.hischool_type_seq.value_counts())
len(resume_education_data.hischool_type_seq.value_counts())

21    6277
0     1724
16     147
8      112
19      92
20      51
22      36
2       17
18      17
23       5
12       3
10       1
Name: hischool_type_seq, dtype: int64


12

In [75]:
print(resume_education_data.hischool_special_type.value_counts())
len(resume_education_data.hischool_special_type.value_counts())

일반고     5820
기타      1724
자율고      501
특성화고     281
특목고      156
Name: hischool_special_type, dtype: int64


5

In [76]:
print(resume_education_data.hischool_nation.value_counts())
len(resume_education_data.hischool_nation.value_counts())

사립    3765
공립    2984
기타    1733
Name: hischool_nation, dtype: int64


3

In [77]:
print(resume_education_data.hischool_gender.value_counts())
len(resume_education_data.hischool_gender.value_counts())

남여공학    2607
여자학교    2423
남자학교    1728
기타      1724
Name: hischool_gender, dtype: int64


4

In [78]:
# print(resume_language_data.resume_seq.value_counts())
# print(resume_language_data.language.value_counts())
print(resume_language_data.exam_name.value_counts())

4     567
11     95
3      55
5      52
2      33
9      25
6      16
1      13
24      6
8       2
21      1
10      1
13      1
7       1
31      1
Name: exam_name, dtype: int64


In [79]:
print(sorted(list(set(resume_data.job_code_seq1))))
print(sorted(list(set(resume_data.job_code_seq2))))
print(sorted(list(set(resume_data.job_code_seq3))))
print(len(list(set(resume_data.job_code_seq1))), len(list(set(resume_data.job_code_seq2))), len(list(set(resume_data.job_code_seq3))))

display(resume_data[(resume_data.job_code_seq1 == '경영·기획·회계·사무') & (resume_data.job_code_seq2 == '재료·화학·섬유·의복') & (resume_data.job_code_seq3 == '영업·판매·TM')].tail(3))
display(resume_data[(resume_data.job_code_seq1 == '경영·기획·회계·사무') & (resume_data.job_code_seq3 == '재료·화학·섬유·의복') & (resume_data.job_code_seq2 == '영업·판매·TM')].tail(3))

['IT·게임', '건설·기계·전기·전자', '경영·기획·회계·사무', '기타 직종', '디자인', '문화·예술·신문·방송', '영업·판매·TM', '재료·화학·섬유·의복', '전문·교육·자격']
['IT·게임', 'NaN', '건설·기계·전기·전자', '경영·기획·회계·사무', '기술·과학·산업', '기타 직종', '디자인', '문화·예술·신문·방송', '여행사', '영업·판매·TM', '재료·화학·섬유·의복', '전문·교육·자격']
['IT·게임', 'NaN', '건설·기계·전기·전자', '관광', '기술·과학·산업', '기타 직종', '디자인', '문화·예술·신문·방송', '영업·판매·TM', '재료·화학·섬유·의복', '전문·교육·자격', '통신·모바일']
9 12 12


resume_seq    reg_date updated_date  degree  graduate_date  hope_salary  \
8217     U02423  2014-01-07   2020-08-07       4           2003       6250.0   
8223     U06838  2012-03-29   2020-10-01       5           1997          0.0   
8288     U01260  2020-01-19   2020-01-19       4           2001          0.0   

      last_salary                                       text_keyword  \
8217       8500.0  생산관리;패턴;의류;우븐;원단;섬유;QC;생산;니트디자인;섬유디자인;텍스타일디자인;...   
8223       3900.0  구매관리;물류관리;물류관리;식음료;식품;식품가공;식품영업;유통영업;고객관리;채널영업...   
8288          0.0           무역;바이어관리;MD;의류;수출입;해외영업;해외구매;무역중개인;스포츠웨어   

     job_code_seq1 job_code_seq2 job_code_seq3  career_month career_job_code  
8217   경영·기획·회계·사무   재료·화학·섬유·의복      영업·판매·TM           120     섬유;봉제;가방;의류  
8223   경영·기획·회계·사무   재료·화학·섬유·의복      영업·판매·TM           120     신문;서적;출판;인쇄  
8288   경영·기획·회계·사무   재료·화학·섬유·의복      영업·판매·TM             0     섬유;봉제;가방;의류

resume_seq    reg_date updated_date  degree  graduate_date  hope_salary  \
7563     U07320  2015-08-01   2020-10-13       4           2003       2700.0   
7775     U02660  2013-02-28   2020-11-22       3           1987       3900.0   
8107     U04811  2017-05-16   2020-04-25       4           2006       3100.0   

      last_salary                                       text_keyword  \
7563       2700.0                              백화점;영업;매장관리;의류;할인점;마트   
7775       4100.0                          머천다이저;바이어관리;바이어상담;MD;해외영업   
8107       2700.0  관리;해외영업;의복;섬유;니트;머천다이저;백화점MD;홈쇼핑MD;화장품MD;MD;무역...   

     job_code_seq1 job_code_seq2 job_code_seq3  career_month career_job_code  
7563   경영·기획·회계·사무      영업·판매·TM   재료·화학·섬유·의복            46     섬유;봉제;가방;의류  
7775   경영·기획·회계·사무      영업·판매·TM   재료·화학·섬유·의복           248     섬유;봉제;가방;의류  
8107   경영·기획·회계·사무      영업·판매·TM   재료·화학·섬유·의복            33     섬유;봉제;가방;의류

In [80]:
print(len(resume_data))
print(resume_data.value_counts(resume_data.job_code_seq1 != ' '))

8482
job_code_seq1
True    8482
dtype: int64
